src/plugins/wizard/public/application/utils/use/use_can_save.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useTypedSelector } from '../state_management';
+
+export const useCanSave = () => {
+  let errorMsg = '';
+  const isEmpty = useTypedSelector(
+    (state) => state.visualization.activeVisualization?.aggConfigParams?.length === 0
+  );
+  const hasChange = useTypedSelector((state) => state.metadata.editorState.state === 'dirty');
+  const hasDraftAgg = useTypedSelector(
+    (state) => !!state.visualization.activeVisualization?.draftAgg
+  );
+
+  // TODO: Need to finalize the error messages
+  if (isEmpty) {
+    errorMsg = 'The canvas is empty. Add some aggregations before saving.';
```

Comment:
do we need to ping @KrooshalUX  to get the final messages?

Reply:
@KrooshalUX Hi Kroosh, for the save button tooltip() error messages for wizard, have we finalized them? Thanks!

Reply:
@btzeng - do we have an empty state for this view?